pkg/repository/git.go

```diff
@@ -100,6 +119,42 @@ func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
 	return s.isClean, s.err
 }
 
+// isClean checks if the repository have changes over the commit
+func (r gitRepoController) isBuildContextClean(ctx context.Context, buildContext string) (bool, error) {
```

Comment:
Same as previous comment about `buildContext`. gitRepo doesn't know what is a `buildContext`. Let's use something that is into `gitrepo` model

Reply:
cf69c7a86197f9bcf2929d9ca9aa1940200334ea